### PR TITLE
Feature: HTML email template

### PR DESCRIPTION
**Pull request by asharma991**
### Summary

#368

This PR implements HTML templates for sending out emails to the users of LogoExecutive. Currently, we're just sending plain emails with very little content. So, this change will improve the communication and brand image.

### Approach

Multiple approaches were taken to create the templates. I made plain HTML templates using some CSS. However, after testing them on various clients, it was observed that the behavior of plain HTML structure is quite different for different clients. So to make it consistent, table structure is used and instead of using classes inline styles are used. (A good guide for creating HTML templates: https://www.freecodecamp.org/news/how-to-create-a-responsive-html-email-template/)

### How to Test the Changes

Try sending out an email using operator credentials or click on forgot password. You'll receive the email in the new templates

### Screenshots or Recordings

These are the screenshots taken for multiple clients

**Before**:
image
image
image

**After**:
image
image
image

## Checklist

- [x] I have added/updated tests that cover the changes.
- [x] I have updated the documentation to reflect the changes.

### CODE REVIEW

**Comment by aps08** [comment]
Can we get the latest screenshots ? There were changes suggested in the meeting. @asharma991

**Review by aps08 — CHANGES_REQUESTED** [review]
HTML templates should be in .html files not directly inside the code. You should import these html files and email the user.

**Comment by asharma991** [comment]
> HTML templates should be in .html files not directly inside the code. You should import these html files and email the user.

But these files have URLs and dynamic content that comes from those respective utilities. Should I create separate functions for these?

**Comment by aps08** [comment]
@asharma991 updates ?

File: server/__tests__/unit/utils/sendEmail.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ describe("sendEmail", () => {
       to: "test@example.com",
       subject: "Test Subject",
       text: "Test Body",
+      html: "Test Body",
```
Comment (Ayushsanjdev) [inline server/__tests__/unit/utils/sendEmail.js:39]: Is this test data required?

File: server/__tests__/unit/utils/sendEmail.js [hunk 2]
```diff
@@ -71,6 +72,7 @@ describe("sendEmail", () => {
...
```
Comment (Ayushsanjdev) [inline server/__tests__/unit/utils/sendEmail.js:75]: +1

File: server/controllers/auth/forgot-password.js [hunk 3]
```diff
@@ -46,8 +43,18 @@ async function forgotPasswordController(req, res, next) {
...
-    const mail = mailText(userToken.tokenURL());
-    const nodeMailerRes = await sendEmail(user.email, mail.subject, mail.body);
+    const htmlFile = fs.readFileSync(
+      __dirname + "/../../templates/ForgotPasswordOrVerify.html",
+      "utf-8"
+    ).toString();
+    const template = handlebars.compile(htmlFile);
+    const replacements = {
+      url: userToken.tokenURL(),
+      message1: "It seems like you forgot your password?",
```
Comment (Ayushsanjdev) [inline server/controllers/auth/forgot-password.js]: Let make text dynamic or add in contants.

File: server/package.json [hunk 4]
```diff
@@ -1,39 +1,40 @@
...
+  },
+  "devDependencies": {
+    "@shelf/jest-mongodb": "^4.3.2",
+    "@types/jest": "^29.5.6",
+    "eslint": "^8.46.0",
+    "jest": "^29.6.4",
+    "nodemon": "^3.0.1",
+    "supertest": "^6.3.3"
+  }
+}
```
Comment (Ayushsanjdev) [inline server/package.json:40]: What is the change in package.json

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]
If this PR is not urgent, you can do the changes.
Rest looks good to me

File: server/__tests__/unit/utils/sendEmail.js (hunk 1 above)
Comment (asharma991) [reply server/__tests__/unit/utils/sendEmail.js:39]: Yup it is, we've added the templates to send the HTML response hence we need this

File: server/package.json (hunk 4 above)
Comment (asharma991) [reply server/package.json:40]: A new dependency handlebars is added for replacements in the HTML templates

File: server/controllers/auth/forgot-password.js (hunk 3 above)
Comment (asharma991) [reply server/controllers/auth/forgot-password.js]: This is only used in this file. Moving it to constants will make sense only when this is used at several places

File: server/controllers/auth/signup.js [hunk 5]
```diff
@@ -106,10 +108,21 @@ async function signupController(req, res, next) {
...
       );
 
+    const htmlFile = fs.readFileSync(
+      __dirname + "/../../templates/ForgotPasswordOrVerify.html",
+      "utf-8"
+    ).toString();
+    const template = handlebars.compile(htmlFile);
+    const replacements = {
+      url: verificationToken.tokenURL(),
+      message1: "You need to verify your email",
```
Comment (aps08) [inline server/controllers/auth/signup.js]: Can we change `message1` and `message2` to better variable names

**Review by aps08 — CHANGES_REQUESTED** [review]
nit

**Review by aps08 — APPROVED** [review]

**Pull request merged**